mysys/my_rdtsc.c

```diff
@@ -175,7 +175,11 @@ ulonglong my_timer_microseconds(void)
 
 ulonglong my_timer_milliseconds(void)
 {
-#if defined(HAVE_SYS_TIMEB_H) && defined(HAVE_FTIME)
+#if defined(HAVE_CLOCK_GETTIME)
+  struct timespec tp;
+  clock_gettime(CLOCK_MONOTONIC_COARSE, &tp);
+  return (ulonglong)tp.tv_sec * 1000 + (ulonglong)tp.tv_nsec / 1000;
```

Comment:
Nanoseconds divided by 1000 became microseconds. I think you should divide by 1000000 here.